packages/frontend/src/diagram/document.ts

```diff
@@ -69,9 +70,13 @@ function enlivenDiagramDocument(
     const { doc } = liveDoc;
 
     const formalJudgments = createMemo<Array<DiagramJudgment>>(() => {
-        return doc.notebook.cells
-            .filter((cell) => cell.tag === "formal")
-            .map((cell) => cell.content);
+        return (
+            doc.notebook.cellOrder
+                // biome-ignore lint/style/noNonNullAssertion:
```

Comment:
Unfortunately I'm adding a lot of these since we now have to do a lookup on an object.

The only thing to do to avoid these assertions is to have an explicit check and throw an error if it fails, however that is a lot more verbose than just letting it fail from being null, especially since we are not expecting this to fail.

The circumstances are different enough each time that there are never more than two examples of the same access pattern, so I didn't think it warranted new helper functions.

It feels messy, and I'm open to other ideas.

Reply:
Yeah, the extra indirection is inherently a little annoying, but in this case and the one below, couldn't you avoid the lint override by writing:

```typescript
doc.notebook.cellOrder
                .map((cellId) => doc.notebook.cellContents[cellId])
                .filter((cell) => cell?.tag === "formal")
```